wom/src/main/scala/wom/types/WomType.scala

```diff
@@ -33,6 +33,7 @@ trait WomType {
       case _ if !coercion.isDefinedAt(any) => Failure(new IllegalArgumentException(
         s"No coercion defined from '${ScalaRunTime.stringOf(any, 3)}' of type" +
           s" '${Option(any.getClass.getCanonicalName).getOrElse(any.getClass.getName)}' to '$toDisplayString'."))
+      case WomOptionalValue(_, Some(v)) => coerceRawValue(v)
```

Comment:
Pulled all the optional handling into `WomType.coerceRawValue` so it didn't need to be duplicated in the subtypes.